larpix/configuration/configuration_v2b.py

```diff
@@ -0,0 +1,330 @@
+    num_bits = 2048
+
+    _endian = 'little'
+    # Additional class properties regarding configuration registers are set at the end of the file.
+
+    def __init__(self):
+        # Note: properties, getters and setters are constructed after this class definition at the bottom of the file.
+        super(Configuration_v2b, self).__init__()
+        return
+
+    def all_data(self, endian='little'):
+        bits = []
+        for register_name in self.register_names:
+            register_data =  getattr(self, register_name+'_data')
+            for register_addr, register_bits in register_data:
+                if len(bits) == register_addr:
+                    if endian[0] == 'l':
+                        bits += [register_bits]
+                    else:
+                        bits += [register_bits[::-1]]
+        return bits
+
+    def some_data(self, registers, endian='little'):
+        '''
+        Fetch register addresses and data from a selected set of registers
+
+        :param registers: list of registers to fetch data for, specified either by register name (str) or register addresses (int)
+
+        :returns: tuple of list of register addresses and list of register data
+
+        '''
+        bits = []
+        addrs = []
+        for register in registers:
+            if isinstance(register, int):
+                register_name = self.register_map_inv[register][0]
+                register_data = getattr(self, register_name+'_data')
+            elif isinstance(register, str):
+                register_data = getattr(self, register+'_data')
+            for register_addr, register_bits in register_data:
+                if isinstance(register, int) and register_addr != register:
+                    continue
+                if endian[0] == 'l':
+                    bits += [register_bits]
+                    addrs += [register_addr]
+                else:
+                    bits += [register_bits[::-1]]
+                    addrs += [register_addr]
+        return addrs, bits
+
+    def from_dict_registers(self, d, endian='little'):
+        '''
+        Load in the configuration specified by a dict of (register,
+        value) pairs.
+
+        '''
+        for address, value in d.items():
+            register_names = self.register_map_inv[address]
+            for register_name in register_names:
+                setattr(self, register_names[0] + '_data', (address, bah.fromuint(value,8,endian=endian)))
+        return
+
+    def _is_register_value_pair(self, item):
+        '''
+        Helper function to determine if item is a register, bitarray pair
+
+        '''
+        if isinstance(item, tuple) and len(item) == 2 and \
+                isinstance(item[0], int) and isinstance(item[1], bitarray)\
+                and len(item[1]) == 8 and item[0] >= 0 and item[0] < self.num_registers:
+            return True
+        return False
+
+## Set up property info
+#
+_property_configuration = OrderedDict([
```

Comment:
You might also consider using the `_property_configuration` declared in configuration_v2.py as a base. I suspect that there is a lot of overlap between the configuration registers, so you should only need to add a couple properties (see configuration_lightpix_v1.py as an example)